File: WarehouseManager/src/main/java/hu/elte/WarehouseManager/model/ProductGroup.java
package hu.elte.WarehouseManager.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.Data;

import javax.persistence.*;
import java.util.List;

@Entity
@Data
@Table(name = "PRODUCT_GROUPS")
public class ProductGroup {

    @Id
    @Column(name = "PRODUCT_GROUP_ID")
    private Integer groupID;

    @Column(name = "PRODUCT_GROUP_NAME")
    private String groupName;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "productGroup")
    @JsonIgnore
    private List<Product> products;
}

File: WarehouseManager/warehouse-manager-client/src/app/model/productgroup.ts
export class ProductGroup {
    
    // fields
    product_group_id: number;
    product_group_name: string;

    // constructor
    constructor(
        product_group_id:number,
        product_group_name:string) { 
            this.product_group_id = product_group_id; 
            this.product_group_name = product_group_name;
    }
}